compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -72,6 +72,19 @@ trait CommonScalaSettings { self: Settings.SettingGroup =>
 }
 
 class ScalaSettings extends Settings.SettingGroup with CommonScalaSettings {
+  private val minTargetVersion = 8
+  private val maxTargetVersion = 15
+
+  private def supportedTargetVersions: List[String] =
+    (minTargetVersion to maxTargetVersion).toList.map(_.toString)
+
+  protected def supportedReleaseVersions: List[String] =
+    if scala.util.Properties.isJavaAtLeast("9") then
+      val jdkVersion = scala.util.Properties.javaSpecVersion.stripPrefix("1.").toInt
```

Comment:
In DirectoryClassPath we use `JDK9Reflectors.runtimeVersionMajor(JDK9Reflectors.runtimeVersion()).intValue()`, I think it would be better to use that everywhere since that's the recommended way to get the version on Java 9+.